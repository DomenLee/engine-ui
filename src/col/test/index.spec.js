import Col from '..';
import Row from '../../row';
import { mount } from '../../../test';

test('Col click event', () => {
  const wrapper = mount(Col);
  wrapper.trigger('click');

  expect(wrapper.emitted('click')).toBeTruthy();
});

test('Row click event', () => {
  const wrapper = mount(Row);
  wrapper.trigger('click');

  expect(wrapper.emitted('click')).toBeTruthy();
});

test('gutter prop', () => {
  const wrapper = mount({
    template: `
      <ghb-row gutter="24">
        <ghb-col span="24">24</ghb-col>

        <ghb-col span="12">12</ghb-col>
        <ghb-col span="12">12</ghb-col>

        <ghb-col span="8">8</ghb-col>
        <ghb-col span="8">8</ghb-col>
        <ghb-col span="8">8</ghb-col>

        <ghb-col span="6">6</ghb-col>
        <ghb-col span="6">6</ghb-col>
        <ghb-col span="6">6</ghb-col>
        <ghb-col span="6">6</ghb-col>
        
        <ghb-col span="7">7</ghb-col>
        <ghb-col span="6">6</ghb-col>
        <ghb-col span="5">5</ghb-col>
        <ghb-col span="4">4</ghb-col>
        <ghb-col span="3">3</ghb-col>
        <ghb-col span="2">2</ghb-col>
      </ghb-row>
    `,
  });

  expect(wrapper).toMatchSnapshot();
});
