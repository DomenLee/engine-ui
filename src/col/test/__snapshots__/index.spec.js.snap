// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gutter prop 1`] = `
<div class="ghb-row">
  <div class="ghb-col ghb-col--24">24</div>
  <div class="ghb-col ghb-col--12" style="padding-right: 12px;">12</div>
  <div class="ghb-col ghb-col--12" style="padding-left: 12px;">12</div>
  <div class="ghb-col ghb-col--8" style="padding-right: 16px;">8</div>
  <div class="ghb-col ghb-col--8" style="padding-left: 8px; padding-right: 8px;">8</div>
  <div class="ghb-col ghb-col--8" style="padding-left: 16px;">8</div>
  <div class="ghb-col ghb-col--6" style="padding-right: 18px;">6</div>
  <div class="ghb-col ghb-col--6" style="padding-left: 6px; padding-right: 12px;">6</div>
  <div class="ghb-col ghb-col--6" style="padding-left: 12px; padding-right: 6px;">6</div>
  <div class="ghb-col ghb-col--6" style="padding-left: 18px;">6</div>
  <div class="ghb-col ghb-col--7" style="padding-right: 18px;">7</div>
  <div class="ghb-col ghb-col--6" style="padding-left: 6px; padding-right: 12px;">6</div>
  <div class="ghb-col ghb-col--5" style="padding-left: 12px; padding-right: 6px;">5</div>
  <div class="ghb-col ghb-col--4" style="padding-left: 18px;">4</div>
  <div class="ghb-col ghb-col--3" style="padding-right: 12px;">3</div>
  <div class="ghb-col ghb-col--2" style="padding-left: 12px;">2</div>
</div>
`;
